screeninfo/enumerators/windows.py

```diff
@@ -64,28 +64,38 @@ def callback(monitor: T.Any, dc: T.Any, rect: T.Any, data: T.Any) -> int:
     # multiple monitors have different DPIs.
     ctypes.windll.shcore.SetProcessDpiAwareness(2)
 
+    # Type of HWND resolves to void *
+    # HWND -> HANDLE -> PVOID -> void *
+    # See https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types
+    hwnd = ctypes.c_void_p(None)
     # On Python 3.8.X GetDC randomly fails returning an invalid DC.
     # To workaround this request a number of DCs until a valid DC is returned.
     for retry in range(100):
         # Create a Device Context for the full virtual desktop.
-        dc_full = ctypes.windll.user32.GetDC(None)
-        if dc_full > 0:
+        # On failure, GetDC returns NULL (type: HDC)
+        # HDC eventually resolves to void *
+        # HDC -> HANDLE -> PVOID -> void *
+        # See https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types
+        GetDC = ctypes.windll.user32.GetDC
+        GetDC.restype = ctypes.c_void_p
```

Comment:
Is it necessary to set `restype` by hand?

Reply:
Necessary, probably not. Without this, the type returned with my environment is c_long. It could be checked against 0 instead, but it's a loss of type safety.
From my understanding it's recommended by [ctypes documentation on foreign functions](https://docs.python.org/3/library/ctypes.html#foreign-functions)

I don't think there is any way that `ctypes` could determine the types for these methods. You can check what information is embedded in the dlly by checking the `dumpbin.exe /EXPORTS` output for `user32.dll`